include/revng/Pipes/PipelineManager.h

```diff
@@ -126,7 +130,7 @@ class PipelineManager {
 
   /// A helper function used to produce all possible targets. It is used for
   /// debug purposes to see if any particular target crashes.
-  llvm::Error produceAllPossibleTargets();
+  llvm::Error produceAllPossibleTargets(bool ExpandTargets = true);
```

Comment:
Mmmh, I'm not a fan of `bool` args, in particular if they have a single user, in particular if they have a default value, in particular if the default value is different from the one of the single user.

Do we really need this?
Can we make a method with a different, more eloquent name?

Reply:
i can do that, but this is only used to implement --produce-all and --produce-all-single, and would introduce a copy of that function just to have a argument less. 

i would prefer less code duplication and keep it this way this time around

Reply:
You can factor the implementation through an `...Impl` method.
I'd suggest to have `--produce-all-individually` and `produceAllPossibleTargetsIndividually`.
The fact that `ExpandTargets` means to produce everything individually is not very intuitive and the fact that is defaulted to `true` is quite strange too.

What's more informative for the reader?

```c++
produceAllPossibleTargets();
produceAllPossibleTargetsIndividually();
```

or

```c++
produceAllPossibleTargets(false);
produceAllPossibleTargets();
```